cardano-explorer/src/Explorer/Web/ClientTypes.hs

```diff
@@ -0,0 +1,264 @@
+    { cbrBlocks     :: [CBlockSummary]
+    , cbrTransactions :: [CTxSummary]
+    } deriving (Show, Generic)
+
+-- | Data displayed on block summary page
+data CBlockSummary = CBlockSummary
+    { cbsEntry      :: !CBlockEntry
+    , cbsPrevHash   :: !CHash
+    , cbsNextHash   :: !(Maybe CHash)
+    , cbsMerkleRoot :: !CHash
+    } deriving (Show, Generic)
+
+data CAddressType
+    = CPubKeyAddress
+    | CScriptAddress
+    | CRedeemAddress
+    | CUnknownAddress
+    deriving (Show, Generic)
+
+data CAddressSummary = CAddressSummary
+    { caAddress :: !CAddress
+    , caType    :: !CAddressType
+    , caTxNum   :: !Word
+    , caBalance :: !CCoin
+    , caTxList  :: ![CTxBrief]
+    } deriving (Show, Generic)
+
+data CTxBrief = CTxBrief
+    { ctbId         :: !CTxId
+    , ctbTimeIssued :: !(Maybe POSIXTime)
+    , ctbInputs     :: ![Maybe (CAddress, CCoin)]
+    , ctbOutputs    :: ![(CAddress, CCoin)]
+    , ctbInputSum   :: !CCoin
+    , ctbOutputSum  :: !CCoin
+    } deriving (Show, Generic)
+
+data CUtxo = CUtxo
+    { cuId       :: !CTxId
+    , cuOutIndex :: !Int
+    , cuAddress  :: !CAddress
+    , cuCoins    :: !CCoin
+    }
+    | CUtxoUnknown
+    { cuTag  :: !Int
+      , cuBs :: !CByteString
+    }
+    deriving (Show, Generic)
+
+newtype CNetworkAddress = CNetworkAddress Text
+    deriving (Show, Generic)
+
+data CTxSummary = CTxSummary
+    { ctsId              :: !CTxId
+    , ctsTxTimeIssued    :: !(Maybe POSIXTime)
+    , ctsBlockTimeIssued :: !(Maybe POSIXTime)
+    , ctsBlockHeight     :: !(Maybe Word)
+    , ctsBlockEpoch      :: !(Maybe Word64)
+    , ctsBlockSlot       :: !(Maybe Word16)
+    , ctsBlockHash       :: !(Maybe CHash)
+    , ctsRelayedBy       :: !(Maybe CNetworkAddress)
+    , ctsTotalInput      :: !CCoin
+    , ctsTotalOutput     :: !CCoin
+    , ctsFees            :: !CCoin
+    , ctsInputs          :: ![Maybe (CAddress, CCoin)]
+    , ctsOutputs         :: ![(CAddress, CCoin)]
+    } deriving (Show, Generic)
+
+data CGenesisSummary = CGenesisSummary
+    { cgsNumTotal               :: !Int
+    , cgsNumRedeemed            :: !Int
+    , cgsNumNotRedeemed         :: !Int
+    , cgsRedeemedAmountTotal    :: !CCoin
+    , cgsNonRedeemedAmountTotal :: !CCoin
+    } deriving (Show, Generic)
+
+data CGenesisAddressInfo = CGenesisAddressInfo
+    { cgaiCardanoAddress :: !CAddress
+    , cgaiGenesisAmount  :: !CCoin
+    , cgaiIsRedeemed     :: !Bool
+    } deriving (Show, Generic)
+
+data CAddressesFilter
+    = RedeemedAddresses
+    | NonRedeemedAddresses
+    | AllAddresses
+    deriving (Show, Generic)
+
+instance Default CAddressesFilter where
+    def = AllAddresses
+
+--------------------------------------------------------------------------------
+-- FromHttpApiData instances
+--------------------------------------------------------------------------------
+
+toCHash :: Hash h a -> CHash
+toCHash = CHash . T.decodeLatin1 . B16.encode . getHash
+
+instance FromHttpApiData CHash where
```

Comment:
Ah, so this is why we need the orphan instances :/ . Maybe the pragma at the top of the file needs a comment.

Reply:
But isn't `CHash` also a local type to the explorer API?

Reply:
i made the instance non-orphaned